org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/NameEnvironment.java

```diff
@@ -203,13 +216,25 @@ private void computeClasspathLocations(
 					}
 					bLocation.patchModuleName = patchedModuleName;
 				} else {
+					String release = Arrays.stream(entry.getExtraAttributes())
```

Comment:
After seeing several new implementations of this lookup, may I suggest `org.eclipse.jdt.internal.core.ClasspathEntry.getExtraAttribute(IClasspathEntry, String)`? :)

Reply:
As we have `org.eclipse.jdt.core.IClasspathEntry.isTest()` I could of course ad something similar for release as well, I just try to keep the API footprint low, just let me know and I'll add something.